site/docs/master/run-locally.md

```diff
@@ -0,0 +1,110 @@
+# Run Velero locally in development
+
+Running the Velero server locally can speed up iterative development. This eliminates the need to rebuild the Velero server
+image and redeploy it to the cluster with each change.
+
+## Run Velero locally with a remote cluster
+
+Velero runs against the Kubernetes API server as the endpoint (as per the `kubeconfig` configuration), so both the Velero server and client use the same `client-go` to communicate with Kubernetes. This means the Velero server can be run locally just as functionally as if it was running in the remote cluster.
+
+### Prerequisites
+
+When running Velero, you will need to ensure that you set up all of the following:
+
+* Appropriate RBAC permissions in the cluster
+  * Read access for all data from the source cluster and namespaces
+  * Write access to the target cluster and namespaces
+* Cloud provider credentials
+  * Read/write access to volumes
+  * Read/write access to object storage for backup data
+* A [BackupStorageLocation][20] object definition for the Velero server
+* (Optional) A [VolumeSnapshotLocation][21] object definition for the Velero server, to take PV snapshots
+
+### 1. Create a cluster
+
+To provision a cluster on AWS using Amazon’s official CloudFormation templates, here are two options:
+
+* EC2 [Quick Start for Kubernetes][17]
+
+* eksctl - [a CLI for Amazon EKS][18]
+
+#### 2. Set enviroment variables
+
+Set the appropriate environment variable for your cloud provider:
+
+AWS: [AWS_SHARED_CREDENTIALS_FILE][15]
+
+GCP: [GOOGLE_APPLICATION_CREDENTIALS][16]
+
+Azure:
```

Comment:
on Azure you can actually set `AZURE_CREDENTIALS_FILE` to the file created at the bottom of this section: https://velero.io/docs/v1.1.0/azure-config/#create-service-principal